doc/gdb-debugging.md

```diff
@@ -0,0 +1,54 @@
+### GDB Debugging
+This document describes the options available for debugging node using the GNU debugger.
+
+We don't provide the debuginfo package, which contains the symbols required for debugging, as this would make our runtime image about 480MB larger. On systems where root access is allowed then the debuinfo package can simply be installed using yum/rpm, but on others like OpenShift getting a pod to run with root access can be somewhat tricky. 
+
+### OpenShift
+When running an application on OpenShift the pod will not have root access. It is possible to configure such access but that means configuring the OpenShift environment. This section provides an alternative to that which allows debugging the application using a remote gdb session. This is done by starting a container in the same cluster with an image with the node runtime and debuginfo, and then connecting to the pod that is the target of the debugging.
+
+#### Start/attach to the running process
+First step is to start the gdbserver on the pod that is the target of the debugging session:
+```
+$ oc rsh nodejs-rest-http-1-nw7hz
+$ gdbserver 172.17.0.6:7777 --attach 23
+Attached; pid = 23
+Listening on port 7777
+```
+In the above case we are creating a new process, but this could also just be the process id of a process running on the pod.
```

Comment:
I think this could be a little clearer. Specifically, is `23` in this case just a random `PID`? Can it be the process id of any process? Is it required that we find the process id of the running node application and use that? It's not really clear from this text.

Reply:
Yeah, I agree. I'm going to go through this today and update it and also make it more realistic with regards to how one would debug the running application and not just setting a breakpoint in the main function (which would never be hit in this case). I mainly wanted to have something basic and see if this was a valid approach before spending/wasting time on it.